system-x/services/ftp/src/main/java/software/tnb/ftp/sftp/service/SFTP.java

```diff
@@ -58,15 +59,21 @@ public SFTPValidation validation() {
     }
 
     public Map<String, String> containerEnvironment() {
-        return Map.of(
-            "SFTP_SERVER_PORT", String.valueOf(port()),
-            "SFTP_SERVER_USERNAME", account().username(),
-            "SFTP_SERVER_PASSWORD", account().password(),
-            "SFTP_SERVER_HOME", account().homeDir()
-        );
+        Map<String, String> env = new HashMap<>();
+        env.put("SFTP_SERVER_PORT", String.valueOf(port()));
+        env.put("SFTP_SERVER_USERNAME", account().username());
+        env.put("SFTP_SERVER_PASSWORD", account().password());
+        env.put("SFTP_SERVER_HOME", account().homeDir());
+
+        // Add TrustedUserCAKeys if configured
+        if (account().trustedUserCAKeys() != null && !account().trustedUserCAKeys().isEmpty()) {
```

Comment:
how would the keys get into the account before the service is created?

I think you should rather make the sftp service configurable and configure the keys that way